launcher/src/main/java/org/apache/brooklyn/launcher/BrooklynWebServer.java

```diff
@@ -447,7 +450,15 @@ public synchronized void start() throws Exception {
         rootContext.setTempDirectory(Os.mkdirs(new File(webappTempDir, "war-root")));
 
         server.setHandler(handlers);
-        server.start();
+        try {
+            server.start();
+        } catch (BindException e) {
+            // port discovery routines may take some time to clear, e.g. 250ms for SO_TIMEOUT
+            // tests fail because of this; see if adding a delay improves things
+            log.warn("Initial server start-up failed binding (retrying after a delay): "+e);
+            Time.sleep(Duration.millis(500));
+            server.start();
```

Comment:
Another possible cause would be parallel tests taking the port just after the check. For tests we could change the logic (here and elsewhere jmx?) to let sockets pick their own listening port.